### ISSUE TITLE
Scala 2.13 with Zinc 1.4.3 fails with NoSuchMethodError

### ISSUE THREAD

**Issue by marc-szalkiewicz-dd** [gradle/gradle#15491]
### Expected Behavior

Gradle's Scala plugin is incompatible with Zinc 1.4.3 and Scala 2.13.

### Current Behavior

A `NoSuchMethod` exception is thrown:

```
Caused by: java.lang.NoSuchMethodError: 'void sbt.internal.inc.AnalyzingCompiler$.compileSources(scala.collection.Iterable, java.io.File, scala.collection.Iterable, java.lang.String, sbt.internal.inc.RawCompiler, sbt.util.Logger)'
        at org.gradle.api.internal.tasks.scala.ZincScalaCompilerFactory.lambda$getBridgeJar$1(ZincScalaCompilerFactory.java:172)
        at org.gradle.cache.internal.DefaultCacheAccess.useCache(DefaultCacheAccess.java:220)
```

### Context

I encountered this issue while using ScalaFix with [ExplicitResultTypes](https://scalacenter.github.io/scalafix/docs/rules/ExplicitResultTypes.html), which requires binary compatibility between the Scala compiler and ScalaFix.

This occurs because the signature to `compileSources` changed in the second parameter from `File` to `Path` between Zinc 1.3.5 and 1.4.3:

[AnalyzingCompiler#compileSources@1.3.5](https://github.com/sbt/zinc/blob/v1.3.5/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala#L318)

[AnalyzingCompiler#compileSources@1.4.3](https://github.com/sbt/zinc/blob/v1.4.3/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala#L400)

With any luck, this change is all that is needed to address https://github.com/gradle/gradle/issues/12869#issuecomment-734900519.

### Steps to Reproduce 

In a convention plugin, set:

```kotlin
scala {
    zincVersion.set("1.4.3")
}

scalafix {
    configFile.set(file("$rootDir/.scalafix.conf"))
}

dependencies {
    zinc("org.scala-sbt:zinc_2.13:1.4.3")
}
```

### Your Environment

OSX, Gradle 6.7.1

**Comment by siddhuwarrier** [comment]
The older version of Zinc uses a log4j version that is vulnerable to https://github.com/advisories/GHSA-jfh8-c2jp-5v3q -- however, we failed to upgrade our Zinc version for our Scala 2.13 project as a result of this issue. Are there any Zinc experts that might be able to help here?

**Comment by NoamGoren1** [comment]
Same for us, would be great to get support

**Comment by arranzropablo** [comment]
I have same problem as @siddhuwarrier

**Comment by natalia-fpintos** [comment]
Having this same issue, trying to use: Zinc `1.5.8` and Scala `2.12.14`

**Comment by Pavloro** [comment]
The fix is urgent due to the vulnerable dependency Apache Log4j
https://snyk.io/vuln/maven%3Aorg.apache.logging.log4j%3Alog4j-core

**Comment by dubinsky** [comment]
Even if you **were** able to use the very latest version of Zinc - 1.6.0 - the problem doesn't go away: it uses log4j 2.17.0, which  has [known vulnerabilities](https://logging.apache.org/log4j/2.x/). Latest Gradle Plugin Publishing Plugin, for instance, rejects any version of log4j prior to 2.17.1 (why is that plugin probing dependencies of the 'zinc' configuration of the project is an interesting question - but it does).

It is possible to **force** the version of log4j regardless of the Zinc version in use. This is described in the [Gradle blog post](https://blog.gradle.org/log4j-vulnerability) (which hasn't been updated and still uses 2.17.0 as the 'good' version ;))

Adopted to Zinc, the stanza to put into the `dependencies` block looks like this:
```groovy
  constraints {
    zinc("org.apache.logging.log4j:log4j-core") {
      version {
        strictly("[2.17.1, 3[")
        prefer("2.17.1")
      }
      because("Log4j vulnerable to remote code execution and other critical security vulnerabilities")
    }
  }
```
With that being said, it would be **really nice** to be able to move past Zinc 1.3.5, which is the latest version supported by Gradle. @tgodzik, is this still planned? https://github.com/gradle/gradle/pull/18001#issuecomment-897028867
> For sure we should update Zinc as the next step. ... I will most likely follow up with that.

**Comment by tgodzik** [comment]
Thanks for reminding me @dubinsky ! I am in the process of updating Zinc in https://github.com/scalacenter/bloop so I should be able to update it here also later on.

**Comment by dubinsky** [comment]
> I should be able to update it here also later on.

Thanks!

**Comment by nikitsenka** [comment]
Any plans to fix this?

**Comment by tgodzik** [comment]
I will take a look next week.

**Comment by tgodzik** [comment]
I think I managed to fix it here https://github.com/gradle/gradle/pull/19864

**Comment by ljacomet** [comment]
PR has been merged and will be released with Gradle 7.5

Thanks again @tgodzik for the changes and others for your reviews and comments!
